woocommerce/woocommerce
# Add notice to connect to woocommerce.com in wc settings and marketplace

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR adds an admin notice for a store that has no woo.com connection, nudging them to connect to woo.com in these location:
- woocommerce settings admin page (achieved by using wp `add_admin_notice`)
- woocommerce marketplace page (achieved by adding a react component)



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Follow the [repo's Getting Started](https://github.com/woocommerce/woocommerce?tab=readme-ov-file#getting-started) and [Plugin's Getting Started](https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce#getting-started) first

#### Store is connected scenario
1. Update the code in here: https://github.com/woocommerce/woocommerce/blob/0407569382f5b5319040b60b10d1845927620ee5/plugins/woocommerce/includes/admin/helper/class-wc-helper.php#L2177 do a `return true` early in the function
2. Open the Woocommerce->Settings page, verify that the notice is not shown 
<img width="1053" alt="image" src="https://github.com/woocommerce/woocommerce/assets/1663428/aee2e928-a990-4242-a92b-84ab701eed16">

3. Open the Woocommerce->Extensions page, verify that the notice is not shown 
<img width="1519" alt="image" src="https://github.com/woocommerce/woocommerce/assets/1663428/befb8804-bab8-42c6-b6b3-ae11193040d1">


#### Store is not connected and have no woo.com plugins
1. Update the code in here: https://github.com/woocommerce/woocommerce/blob/0407569382f5b5319040b60b10d1845927620ee5/plugins/woocommerce/includes/admin/helper/class-wc-helper.php#L2177 do a `return false` early in the function
2. Open the Woocommerce->Settings page, verify that the notice is not shown 
<img width="1053" alt="image" src="https://github.com/woocommerce/woocommerce/assets/1663428/aee2e928-a990-4242-a92b-84ab701eed16">

3. Open the Woocommerce->Extensions page, verify that the notice is not shown 
<img width="1519" alt="image" src="https://github.com/woocommerce/woocommerce/assets/1663428/befb8804-bab8-42c6-b6b3-ae11193040d1">


#### Store is not connected and have a woo.com extension that is up to date
1. Update the code in here: https://github.com/woocommerce/woocommerce/blob/0407569382f5b5319040b60b10d1845927620ee5/plugins/woocommerce/includes/admin/helper/class-wc-helper.php#L2177 do a `return false` early in the function
2. Open woo.com, download any paid plugin there (for example: woo subscriptions), install it in the local woocommerce core
3. Open the Woocommerce->Settings page, verify that the notice is shown 
<img width="1034" alt="image" src="https://github.com/woocommerce/woocommerce/assets/1663428/400c98cc-c273-499f-a5a4-6a9b8914bc2b">

4. Open the Woocommerce->Extensions page, verify that the notice is shown 
<img width="1603" alt="image" src="https://github.com/woocommerce/woocommerce/assets/1663428/f11d528f-b36f-4fde-9920-d0403e721b79">

#### Store is not connected and have a woo.com extension that is out of date
1. Update the code in here: https://github.com/woocommerce/woocommerce/blob/0407569382f5b5319040b60b10d1845927620ee5/plugins/woocommerce/includes/admin/helper/class-wc-helper.php#L2177 do a `return false` early in the function
2. Open woo.com, download any paid plugin there (for example: woo subscriptions), install it in the local woocommerce core
3. Open the Woocommerce->Settings page, verify that the longer notice is shown 
<img width="1603" alt="image" src="https://github.com/woocommerce/woocommerce/assets/1663428/809f2a33-519a-45c0-81d6-f868b4575832">


4. Open the Woocommerce->Extensions page, verify that the longer notice is shown 
<img width="1603" alt="image" src="https://github.com/woocommerce/woocommerce/assets/1663428/46ed0808-8691-4f79-91fd-1b0f8f8319fe">



#### Test hyperlink and button
1. Open the Woocommerce->Settings page, verify that clicking the hyperlink "Connect your store" redirect to My subscription page
2. Open the Woocommerce->Extensions page, verify that clicking the "Connect" button initiate a store connection to woo.com

#### Test notice dismissal persistence
1. Verify that dismissing the notice is persisted on the page refresh
2. Delete these following local storage key to make them show again: `woo-connect-notice-settings-dismissed` and `wc-marketplaceNoticeClosed-woo-connect-notice`

#### Test notice re-show after dismissal in > 1 month
1. Dismiss both notices in settings and marketplace page
2. Modify these following 2 local storages, set the date to more than 1 month ago: `woo-connect-notice-marketplace-dismissed` and `woo-connect-notice-settings-dismissed`
3. Verify that refreshing the page shows the notice again
